version: '3.7'

volumes: 
  grafana_data: {}

networks:
  front-tier:
  back-tier:

services:
  db:
    image: postgres
    restart: always
    environment:
      POSTGRES_PASSWORD: <password>
    ports:
      - "5432:5432"
    networks:
      - back-tier

  adminer:
    image: adminer
    restart: always
    ports:
      - "8080:8080"
    networks:
      - back-tier
      - front-tier

  grafana:
    image: grafana/grafana
    user: "472"
    ports:
      - "3001:3000"
    volumes:
      - ./config/grafana_datasources.yaml:/etc/grafana/provisioning/datasources/datasource.yaml:ro
      - ./config/grafana_dashboards.yaml:/etc/grafana/provisioning/dashboards/dashboards.yaml:ro
      - ./dashboards:/opt/grafana/dashboards
    networks:
      - back-tier
      - front-tier
    restart: always

  web:
    build: .
    image: webapp:1.0.0
    extra_hosts:
      - "host.docker.internal:host-gateway"
    ports:
      - "9696:9696"
    environment:
      RUN_ID: ed82613605f34f34bff58591d9f8739d
      TRACKING_URI: http://host.docker.internal:5000
      EXPERIMENT: online-course-engagement-prediction-experiment-1
      DB_HOST: db
      DB_USER: postgres
      DB_PASSWORD: <password>
      DB_PORT: 5432
      DB_NAME: course
    networks:
      - back-tier